### PR TITLE
Filter line break and trailing whitespaces from secrets

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Filter line break and trailing whitespaces from secrets

### Describe how to test/QA your changes

Set a secrets value with trailing spaces and check they were correctly removed.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/secrets/fetch_secret.go [hunk 1]
```diff
@@ -150,15 +150,20 @@ func fetchSecret(secretsHandle []string, origin string) (map[string]string, erro
 		if v.ErrorMsg != "" {
 			return nil, fmt.Errorf("an error occurred while decrypting '%s': %s", sec, v.ErrorMsg)
 		}
-		if v.Value == "" {
+
+		value := strings.TrimSpace(v.Value)
```
Comment (hush-hush) [inline pkg/secrets/fetch_secret.go:154]: Should we trim whitespaces or only line break at the start and end of the secret ?

Comment (truthbk) [reply pkg/secrets/fetch_secret.go:154]: I think both are probably acceptable; we might get some support cases, but I think it'll be a rare occurrence. We'll have to advertise this in the release notes.

**Review by truthbk — APPROVED** [review]
Looks good to me as a short-term fix.

**Comment by hush-hush** [comment]
Closing in favor of https://github.com/DataDog/datadog-agent/pull/15707 and https://github.com/DataDog/datadog-agent/pull/15856

**Pull request closed without merging**